openfpga/src/base/openfpga_build_fabric_template.h

```diff
@@ -142,11 +144,15 @@ int build_fabric_template(T& openfpga_ctx, const Command& cmd,
     }
   }
 
-  if (true == cmd_context.option_enable(cmd, opt_compress_routing)) {
+  if (true == cmd_context.option_enable(cmd, opt_compress_routing) &&
+      false == openfpga_ctx.device_rr_gsb().get_is_dirty_flag()) {
```

Comment:
If it is not dirty (it indicates that a preloading is done?), we should NOT run the compress_routing?

Reply:
This is saying that if it is not dirty (no preloading) & compress_routing is on, then we need to call default build_unique_module methods

Reply:
O.k. It is better to rename the flag from ``is_dirty`` to ``is_compressed``. Then the confusion should be resolved.